### PR TITLE
Update docstring links in links.rst and Codebase

### PR DESCRIPTION
**Pull request by XFarooqi**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

This Pull Request will Fix issue #504 and also I replace all the links in Codebase.

## Did you have fun?

Yeah, I had fun finding the links in the docstrings and placing them in links.rst.

### CODE REVIEW

File: torchmetrics/functional/text/bert.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def _get_precision_recall_f1(
 
 
 def _get_hash(model_name_or_path: Optional[str] = None, num_layers: Optional[int] = None, idf: bool = False) -> str:
-    """Copied from https://github.com/Tiiiger/bert_score/blob/master/bert_score/utils.py and adjusted."""
+    r"""BERT_score\_"""
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:379]:
```suggestion
    r"""`BERT score`_"""
```

**Review by Borda — COMMENTED** [review]
Seems that most of the replaced links have some formatting issue, mind see this sample correction?

File: torchmetrics/functional/text/bert.py (hunk 1 above)
Comment (XFarooqi) [reply torchmetrics/functional/text/bert.py:379]: My bad, I will make it correct.

Comment (XFarooqi) [reply torchmetrics/functional/text/bert.py:379]: Besides this, If anything seems missing or needs to be corrected. Let me know

Comment (XFarooqi) [reply torchmetrics/functional/text/bert.py:379]: In Your suggestion, you want r"""`BERT score`_"""


My question is in the link.rst this is mentioned as BERT_score, So Do I need to remove underscore(_) while replacing it in Codebase or write it as it is written in links.rst

**Pull request closed without merging**